pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/BaseJoinOperator.java

```diff
@@ -376,4 +382,126 @@ public StatMap.Type getType() {
       return _type;
     }
   }
+
+  /**
+   * This util class is a view over the left and right row joined together
+   * currently this is used for filtering and input of projection. So if the joined
+   * tuple doesn't pass the predicate, the join result is not materialized into Object[].
+   *
+   * It is debatable whether we always want to use this instead of copying the tuple
+   */
+  private abstract static class JoinedRowView extends AbstractList<Object> implements List<Object> {
```

Comment:
Would it be possible to enhance and re-use this existing util class instead - https://github.com/apache/pinot/blob/60f1897d95024db29b820fdd87c1962809f2e9fb/pinot-common/src/main/java/org/apache/pinot/common/utils/list/FlatViewList.java#L31